### ISSUE TITLE
Do not include time component in startDate and endDate for all-day events

### ISSUE THREAD

**Issue by nickevansuk** [openactive/developer-microsite#40]
## Rule name
Do not include time component in startDate and endDate for all-day events

## Rule type
Data quality warning

## Rule hint
All day events should not include a time component, as they may be interpreted having a start time rather than occurring on a date. 

## Rule notes
Detect `startDate` and `endDate` where the time is set to midnight (either with the timezone stripped, or when converted to UTC), as it's unlikely events that are not all-day start at midnight with zero duration.

## Associated proposal or model reference
- https://github.com/openactive/modelling-opportunity-data/issues/94

## Examples
- https://github.com/openactive/activation/issues/158

**Comment by ldodds** [comment]
I don't think this makes sense. An all day event will have a specific start and end time that a user will want to see. 

The referenced proposal suggests using a combination of extra types, schedule, etc to describe multi day events. A simpler approach would be to distinguish sessions from longer events, which is one of the reasons for adding more custom Event types. I would prefer to go with that option.

**Comment by nickevansuk** [comment]
Did you read the rule note above "Detect startDate and endDate where the time is set to midnight (either with the timezone stripped, or when converted to UTC), as it's unlikely events that are not all-day start at midnight with zero duration", it's specifically to catch "all day" events with a start and end time of midnight, and suggest that the time component is not included.

**Comment by blabyboy** [comment]
How likely is it to have events that start time and end time of midnight?

From my (limited) experience, even Midnight runs etc.. have a start time of 10pm on the previous evening for registration etc.. I would personally like to see a start and end time set on any event even if it is start: 00:00 and end is: 23:59.

**Comment by nickevansuk** [comment]
Totally agree Paul, to clarify the problem: for events that have no time data stored because the system they are stored in doesn’t facilitate a time component (i.e the platform only stores a date), the default for most libraries is render it to ISO spec with the time set to midnight.

Most data consumers will then display this to users as an event starting at midnight, which is inaccurate. As you rightly say it most likely starts at some time in the am. The suggestion here is for the validator to pick up on such incidents and warn the publisher that they may be misleading their audience by accidentally including inaccurate time data.
